uaclient/tests/test_contract.py

```diff
@@ -777,3 +825,51 @@ def fake_contract_client(cfg):
             ),
         ]
         assert process_calls == process_entitlement_delta.call_args_list
+
+
+@mock.patch("uaclient.contract.UAContractClient.get_updated_contract_info")
+class TestContractChanged:
+    @pytest.mark.parametrize("is_expired", (False, True))
+    def test_contract_change_with_expiry(
+        self, get_updated_contract_info, is_expired, FakeConfig
+    ):
+        if is_expired:
+            expiry_date = "2041-05-08T19:02:26Z"
+            ret_val = True
+        else:
+            expiry_date = "2040-05-08T19:02:26Z"
+            ret_val = False
+        get_updated_contract_info.return_value = {
+            "machineTokenInfo": {
+                "contractInfo": {
+                    "effectiveTo": expiry_date,
+                },
+            },
+        }
+        cfg = FakeConfig().for_attached_machine()
+        assert is_contract_changed(cfg) == ret_val
+
+    @pytest.mark.parametrize("ent_changed", (False, True))
+    def test_contract_change_with_entitlements(
+        self, get_updated_contract_info, ent_changed, FakeConfig
+    ):
+        if ent_changed:
+            resourceEntitlements = [{"type": "token1", "entitled": True}]
+            resourceTokens = [{"token": "token1", "type": "resource1"}]
+            ret_val = True
```

Comment:
I don't think we need `ret_val` here, we can simply use `ent_changed` directly

Reply:
Or we can change the variable name to `has_contract_changed`, just to make the test more readable

Reply:
Oh yes, changing the variable name and removing `ret_val`.